### PR TITLE
Enhancement to s3 canary deployments

### PR DESCRIPTION
**Pull request by ndcampbell**
Added a feature so that when `deploy_strategy=canary`, an S3 bucket will deploy to a `CANARY` path as well. 

The versioned path still remains the source of deployment and `CANARY` and `LATEST` get synced from there.

### CODE REVIEW

File: src/foremast/s3/s3deploy.py [hunk 1]
```diff
@@ -68,8 +68,10 @@ def setup_pathing(self):
         s3_format = "s3://{}"
         version_path = path_format.format(self.bucket, self.s3path, self.version).replace('//', '/')
```
Comment (sijis) [inline src/foremast/s3/s3deploy.py]: Somehow, this continous `.replace('//', '/')` seems strange. Maybe there's a way to do this earlier or replace with a function?

Comment (ndcampbell) [inline src/foremast/s3/s3deploy.py]: Yeah, I can refactor this so that a function is doing it

**Comment by ndcampbell** [comment]
@sijis added small doc on this deploy_strategy. We have issue #153 for creating docs on S3 deployments but we do not have anything on that yet. I also renamed those variables

**Comment by sijis** [comment]
All good.

**Pull request merged**